[WIP][Nyzul] Rewrite Nyzul Isle Investigation

Xaver-DaRed opened the pull request:
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [Obviously no] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Originally intended to contain all the rewrite, it became obvious it was far too large and far too complex to be reviewed in a sane manner.

So now, this PR will act as a guide for the smaller PRs that will be opened containing small chunks of the rewrite at a time, to, hopefully, make it simpler to review and to code without having to rebase multiple times.

It will also act as the final PR of the rewrite, couse I already opened it and why clutter the PR history?

## Steps to test these changes

Half of it it's already in, so... Do nyzul and see if it works, I guess, I dunnow. What's nyzul?

Xaver-DaRed committed:
Move all pathos logic to its own file.

- Remove pathos logic and tables from nyzul global.
- Create global functions for gear mobs engage and death.
- Cleanup mob scripts and family mixin.
- Adjust pets.lua pathos lookup in case it gets changed.

Xaver-DaRed committed:
Simplify entity tables

- Renamed table names.
- Addopted a "first mob, last mob" pattern across all tables.
- Changed logic to reflect this changes.
- Renamed how dynamic tables are named, so we can now they are crated and edited on the spot.
- Changed how rates are handled, to make them clearer.

Xaver-DaRed committed:
Raname "NII_pathos" file to just "pathos"

- Rename "NII_pathos" to "pathos". We are inside a folder, we dont need the NII_ subfix anymore.
- Rename addFloorPathos to addPathos

Xaver-DaRed committed:
Unify mob generation logic and data to the same file.

- Move logic from instance file to "floor_generation"
- Move data from nyzul global AND points file to "floor_generation"
- Adjust spawn point data to simplify logic

UmeboshiXI commented:
If you need help testing anything, I'm willing to help out. I was getting ready to tackle this myself for our server. 

Some things of note with the current implementation and the client crash when entering the instance. The client crash can be avoided if you are in the menu when instance leader/initiator confirms entry.

Xaver-DaRed commented:
Once it's ready, feel free to cherry-pick and test it. The more ppl helping testing it, the better.

Xaver-DaRed committed:
Move lamp pos table to floor generation.

- Delete points file entirely.
- Delete references to said file.
- Cleanup logic for lamp positioning

zach2good commented:
When this is eventually ready for review, I think you're going to have to write a little review guide so we know where to look, because this is _massive_ :D

Xaver-DaRed committed:
Start floor progression and variable conversion

Xaver-DaRed committed:
Move all treasure chest and casket logic to it's own file

- Cleanup and comenting mostly.
- Left a couple of TODOs in there

Xaver-DaRed committed:
Convert Rune_of_Transfer_Start
